### DISCUSSION TITLE
Small little glitch

### DISCUSSION THREAD

**Discussion by beaver700nh** [community/community#14901] [Copilot Conversations]
Hello, it's suggesting something... strange
![screenshot_2022-04-13_19-02-55-093060461](https://user-images.githubusercontent.com/41931581/163289134-b9155991-0215-43d0-91dc-0fee1a524e6b.png)

It's a bit funny though lol

Also love this:
https://user-images.githubusercontent.com/41931581/163289156-4f61f65d-a916-4c03-9272-12974b13407a.mp4

**Comment by Kek5chen** [comment]
This is currently happening to multiple users, including me.
I'll just quote myself here from another discussion.
> Now to answer this and the quote from before: Because of those patterns, it could be that a pattern repeats itself resulting in a repeating output. This is not a bug, but more of expected behaviour. Code repetition shouldnt be a thing, and filtered out, though this is still actively in development. A note to GitHub, like you and others here wrote, is the right thing to do to bring this to notice.

> I recommend to give it more context and it'll be more precise for you in return. For example by writing a function signature and letting copilot do the rest.
This behaviour usually happens when all patterns are fullfilled (like: class with getter and setter). What would it complete? It makes something up to speed up your development! I sometimes use this to get ideas on how to solve a certain task.

**Reply by beaver700nh** [reply]
Hmm ok
This was just after I added a `private:` label, and it suggested this. A bit strange because the only reference to paint was a single line above:
```c++
  Status paint();
```

**Reply by beaver700nh** [reply]
what the
wow my wifi sucks

edit: deleted it (accidentally triple posted)

**Reply by Kek5chen** [reply]
If my answer still helped you it would be nice if you could mark it as the answer

**Reply by beaver700nh** [reply]
I would but it doesn't really answer why there's a bunch of repeated `cancel`s and why the last line ends with `confirm_c` (completing neither the line nor the pattern). Giving it more context would work, but that's a workaround. This is still unexpected behavior.

**Reply by boermansjo** [reply]
> I would but it doesn't really answer why there's a bunch of repeated `cancel`s and why the last line ends with `confirm_c` (completing neither the line nor the pattern). Giving it more context would work, but that's a workaround. This is still unexpected behavior.

That is because the AI is not that intelligent. 
It will give you a suggestion that a cancel function will have 3 functions : "yes", "no" and "cancel",
then it will just repeat that for ever because it keeps adding the "cancel" function that again needs 3 functions...
(In the mind of the AI of course)

The end is just a hard stop I guess to prevent any infinite loops. (Luckily it stopped otherwise you would have infinite lines with tons of cancels)

**Comment by baartch** [comment]
😄
<a href="https://ibb.co/1MwDncT"><img src="https://i.ibb.co/gdB57YJ/2022-09-08-09-02-32-Window.png" alt="2022-09-08-09-02-32-Window" border="0"></a>
